src/main/java/org/opentripplanner/routing/core/AStarRequestBuilder.java

```diff
@@ -0,0 +1,120 @@
+package org.opentripplanner.routing.core;
+
+import java.time.Instant;
+import org.opentripplanner.model.GenericLocation;
+import org.opentripplanner.routing.api.request.StreetMode;
+import org.opentripplanner.routing.api.request.preference.RoutingPreferences;
+import org.opentripplanner.routing.api.request.request.VehicleParkingRequest;
+import org.opentripplanner.routing.api.request.request.VehicleRentalRequest;
+
+public class AStarRequestBuilder {
+
+  private Instant startTime = Instant.now();
+  private StreetMode mode = StreetMode.WALK;
+  private RoutingPreferences preferences = new RoutingPreferences();
+  private boolean arriveBy = false;
+  private boolean wheelchair = false;
+  private VehicleParkingRequest parking = new VehicleParkingRequest();
+  private VehicleRentalRequest rental = new VehicleRentalRequest();
+  private GenericLocation from = null;
+  private GenericLocation to = null;
```

Comment:
This works, but I in general prefer to have the default values in a private constructor in the "main" class, since this ir part of the domain logic . Then the builder does not know any business logic and can do one thing, build.

Reply:
How should I handle startTime, which should be the current time in the request, if not explicitly set?